src/lib/logical_query_plan/build_lqps.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <memory>
+
+#include "types.hpp"
+#include "all_parameter_variant.hpp"
+#include "all_type_variant.hpp"
+#include "lqp_column_reference.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class CreateViewNode;
+class InsertNode;
+class LimitNode;
+class LQPExpression;
+class ProjectionNode;
+class PredicateNode;
+class StoredTableNode;
+class UpdateNode;
+
+std::shared_ptr<PredicateNode> make_predicate_node(const LQPColumnReference& column_reference, const PredicateCondition predicate_condition, const AllParameterVariant& value, const std::shared_ptr<AbstractLQPNode>& child);
```

Comment:
Good guess that I won't like it. ;)

I do see value of having those methods though. How do you feel about `PredicateNode::make()`?

Also, I would consider moving the child arguments to the front. Feels more consistent to me.

Reply:
PredicateNode::make would be cool with me, it's not even longer than make_predicate_node.

The reason I put the children last is because this enables makes it possible to read the structure of an LQP from the code that creates it:

```
  const auto expected_optimized_lqp =
  make_predicate_node(_mock_node_b, PredicateCondition::GreaterThan, 40,
    make_predicate_node(_mock_node_b, PredicateCondition::GreaterThan, 50,
      make_predicate_node(_mock_node_a, PredicateCondition::Equals, 42,
        make_star_projection_node(
          make_predicate_node(_mock_node_c, PredicateCondition::LessThan, 500,
            make_predicate_node(_mock_node_a, PredicateCondition::GreaterThanEquals, 90,
              _mock_node
  ))))));
```

I think having this trait is very important - imagine that code with the children first!

Reply:
Yepp, I like it.